File: java/src/main/java/Fibonacci.java
import java.util.ArrayList;

public class Fibonacci {
    public Fibonacci() {

    }

    public ArrayList<Integer> forNumber(Integer sequenceMaximum) {
        ArrayList<Integer> fibonacci = new ArrayList<>();

        for (int i = 0; i < sequenceMaximum; i++) {
            fibonacci.add(fibonacciLoop(i+1));
        }

        return fibonacci;

    }

    public Integer fibonacciLoop(Integer index) {
        if (index == 1 || index == 2) {
            return 1;
        }
        int firstFibo = 1, secondFibo = 1;
        int sum = 0;
        for (int i = 3; i <= index; i++) {
            //Fibonacci number is sum of previous two Fibonacci number
            sum = firstFibo + secondFibo;
            firstFibo = secondFibo;
            secondFibo = sum;

        }
        return sum;
    }
}

File: javascript/src/main/fibonacci.js
fibonacci = function(numberOfSequence) {

    var sequence = [];

    var index = 1;
    while (index <= numberOfSequence) {
        sequence.push(calculateNthNumber(index));
        index++;
    }
    return sequence;
};

calculateNthNumber = function(nthNumber) {

    if (nthNumber === 1 || nthNumber == 2) {
        return 1;
    }

    var firstFibo = 1, secondFibo = 1, sum = 0;
    for (var i = 3; i <= nthNumber; i++) {
        sum = firstFibo + secondFibo;
        firstFibo = secondFibo;
        secondFibo = sum;
    }

    return sum;
};

